### PR TITLE
Add dynamic logging capabilities

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

File: promlog/log.go [hunk 1]
```diff
@@ -106,3 +124,52 @@ func New(config *Config) log.Logger {
...
+}
+
+type logger struct {
+	base         log.Logger
+	leveled      log.Logger
+	currentLevel *AllowedLevel
+	mtx          sync.Mutex
+}
+
+// l implements logger.Log
```
Comment (SuperQ) [inline promlog/log.go:157]: Is this the correct syntax for this godoc?

File: promlog/log.go [hunk 2]
```diff
@@ -106,3 +124,52 @@ func New(config *Config) log.Logger {
...
+func (l *logger) Log(keyvals ...interface{}) error {
+	l.mtx.Lock()
+	defer l.mtx.Unlock()
+	return l.leveled.Log(keyvals...)
+}
+
+// SetLevel allows changing the level
```
Comment (SuperQ) [inline promlog/log.go:164]:
```suggestion
// SetLevel changes the output level
```

File: promlog/log.go (hunk 1 above)
Comment (roidelapluie) [reply promlog/log.go:157]: It is not, but I had fixed it. Where did I push...

**Review by SuperQ — APPROVED** [review]
LGTM

**Pull request merged**